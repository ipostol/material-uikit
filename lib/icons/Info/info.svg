<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24">
    <g fill="none" fill-rule="evenodd" stroke="#9499B0">
        <path stroke-width=".01" d="M.005.005L0 23.995 23.995 24 24 .005H.005z"/>
        <rect width="18.8" height="18.8" x="2.6" y="2.6" stroke-width="1.2" rx="9.4"/>
        <path stroke-linecap="round" stroke-width="1.4" d="M12 11v5"/>
        <path stroke-linecap="round" stroke-width="1.6" d="M12 7.8v.1"/>
    </g>
</svg>
