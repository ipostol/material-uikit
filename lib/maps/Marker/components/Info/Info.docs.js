'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactMarkupComponents = require('react-markup-components');

var _Info = require('./Info');

var _Info2 = _interopRequireDefault(_Info);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = function () {
  return _react2.default.createElement(
    _reactMarkupComponents.Skin,
    null,
    _react2.default.createElement(_Info2.default, {
      timeslotsCount: 3,
      minPrice: 123,
      maxPrice: 321,
      typeIcon: 'wj0hwqmnqnzj9qgvbisy',
      typeName: 'Digital Billboard',
      resolution: '15x50',
      looping: '30s, 30m, 10m, 10m, 30s, 30m, 10m, 10m',
      place: 'Gregg Point'
    })
  );
};