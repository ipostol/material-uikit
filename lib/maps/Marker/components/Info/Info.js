'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _numeral = require('numeral');

var _numeral2 = _interopRequireDefault(_numeral);

var _bidstackCloudinary = require('bidstack-cloudinary');

var _bidstackCloudinary2 = _interopRequireDefault(_bidstackCloudinary);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _Place = require('../../../../icons/Place');

var _Place2 = _interopRequireDefault(_Place);

var _Timeslots = require('../../../../icons/Timeslots');

var _Timeslots2 = _interopRequireDefault(_Timeslots);

var _TimeslotsTooltip = require('../../../../icons/TimeslotsTooltip');

var _TimeslotsTooltip2 = _interopRequireDefault(_TimeslotsTooltip);

var _Resolution = require('../../../../icons/Resolution');

var _Resolution2 = _interopRequireDefault(_Resolution);

var _Looping = require('../../../../icons/Looping');

var _Looping2 = _interopRequireDefault(_Looping);

var _Info = require('./Info.scss');

var _Info2 = _interopRequireDefault(_Info);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Info = function Info(_ref) {
  var timeslotsCount = _ref.timeslotsCount,
      minPrice = _ref.minPrice,
      maxPrice = _ref.maxPrice,
      typeName = _ref.typeName,
      typeIcon = _ref.typeIcon,
      resolution = _ref.resolution,
      looping = _ref.looping,
      place = _ref.place;


  var isMultiple = timeslotsCount > 1;

  var price = isMultiple ? _react2.default.createElement(
    'div',
    null,
    '\xA3',
    (0, _numeral2.default)(minPrice).format('0,0'),
    ' \u2013 \xA3',
    (0, _numeral2.default)(maxPrice).format('0,0')
  ) : _react2.default.createElement(
    'span',
    null,
    '\xA3',
    (0, _numeral2.default)(minPrice).format('0,0')
  );

  var icon = _bidstackCloudinary2.default.url(typeIcon, { width: 16, height: 16 });

  return _react2.default.createElement(
    'div',
    { className: _Info2.default.general },
    _react2.default.createElement(
      'div',
      { className: _Info2.default.header },
      _react2.default.createElement(
        'div',
        { className: _Info2.default.price },
        price
      )
    ),
    _react2.default.createElement('div', { className: _Info2.default.line }),
    _react2.default.createElement(
      'div',
      { className: _Info2.default.infoWrap },
      _react2.default.createElement(_Place2.default, { className: (0, _classnames2.default)(_Info2.default.icon, _Info2.default.invertIcon) }),
      _react2.default.createElement(
        'div',
        null,
        place
      )
    ),
    _react2.default.createElement(
      'div',
      { className: _Info2.default.infoWrap },
      _react2.default.createElement('img', { src: icon, alt: '', className: _Info2.default.icon }),
      _react2.default.createElement(
        'div',
        null,
        typeName
      )
    ),
    _react2.default.createElement(
      'div',
      { className: _Info2.default.infoWrap },
      isMultiple ? _react2.default.createElement(_TimeslotsTooltip2.default, { className: _Info2.default.icon }) : _react2.default.createElement(_Timeslots2.default, { className: (0, _classnames2.default)(_Info2.default.icon, _Info2.default.invertIcon) }),
      _react2.default.createElement(
        'div',
        null,
        timeslotsCount,
        ' Timeslot',
        isMultiple ? 's' : ''
      )
    ),
    _react2.default.createElement(
      'div',
      { className: _Info2.default.infoWrap },
      _react2.default.createElement(_Resolution2.default, { className: (0, _classnames2.default)(_Info2.default.icon, _Info2.default.invertIcon) }),
      _react2.default.createElement(
        'div',
        null,
        resolution,
        'px'
      )
    ),
    looping && _react2.default.createElement(
      'div',
      { className: _Info2.default.infoWrap },
      _react2.default.createElement(_Looping2.default, { className: (0, _classnames2.default)(_Info2.default.icon, _Info2.default.invertIcon) }),
      _react2.default.createElement(
        'div',
        null,
        'Looping ',
        looping
      )
    )
  );
};

Info.propTypes = {
  place: _propTypes2.default.node,
  typeIcon: _propTypes2.default.string,
  typeName: _propTypes2.default.node,
  timeslotsCount: _propTypes2.default.number,
  minPrice: _propTypes2.default.number,
  maxPrice: _propTypes2.default.number,
  resolution: _propTypes2.default.string,
  looping: _propTypes2.default.string
};

exports.default = Info;