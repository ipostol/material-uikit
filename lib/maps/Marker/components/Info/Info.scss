@import '~bidstack-theme/theme';

$tooltipHeader: 15px;

.general {
  padding: 8px;
  box-sizing: border-box;
  width: 200px;
  z-index: 10;
  border-radius: 4px;
  background-color: rgba(0, 0, 0, 0.98);
  color: $alternateTextColor;
}

.header {
  display: flex;
  flex-flow: row wrap;
  align-items: center;
  width: 200px;
}

.wrap {
  display: flex;
  align-items: baseline;
}

.price {
  font-size: $tooltipHeader;
}

.winning {
  position: relative;
  font-size: 12px;
  line-height: 1.33;
  color: #38c52a;
}

.loosing {
  position: relative;
  font-size: 12px;
  line-height: 1.33;
  color: #ff4d4d;
}

.circle {
  display: inline-block;
  margin: 4px;
  margin-left: 8px;
  width: 8px;
  height: 8px;
  border-radius: 10px;
}

.line {
  margin: 8px 0 10px;
  width: 100%;
  height: 1px;
  background-color: rgba(255, 255, 255, 0.08);
}

.winningColor {
  background-color: #38c52a;
}

.loosingColor {
  background-color: #ff4d4d;
}

.icon {
  padding-right: 6px;
  min-width: 16px;
  height: 16px;
}

.invertIcon {
  filter: invert(100%);
}

.infoWrap {
  display: flex;
  align-items: flex-start;
  font-size: 13px;
  font-weight: 500;
  line-height: 1.23;
  color: rgba(255, 255, 255, 0.72);
  margin-top: 8px;
}
