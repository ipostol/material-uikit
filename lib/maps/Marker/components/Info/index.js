'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _Info = require('./Info');

var _Info2 = _interopRequireDefault(_Info);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = _Info2.default;