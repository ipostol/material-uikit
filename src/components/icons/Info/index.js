import React from 'react';

const image = require('./info.svg');

const Info = props => (
  <img src={image} alt="" {...props} />
);

export default Info;
