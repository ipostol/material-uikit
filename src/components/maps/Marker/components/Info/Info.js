import React from 'react';
import PropTypes from 'prop-types';
import numeral from 'numeral';
import cloudinary from 'bidstack-cloudinary';
import cn from 'classnames';
import PlaceIcon from '../../../../icons/Place';
import DateIcon from '../../../../icons/Timeslots';
import TimeslotsTooltipIcon from '../../../../icons/TimeslotsTooltip';
import ResolutionIcon from '../../../../icons/Resolution';
import LoopingIcon from '../../../../icons/Looping';
import styles from './Info.scss';

const Info = ({
  timeslotsCount,
  minPrice,
  maxPrice,
  typeName,
  typeIcon,
  resolution,
  looping,
  place,
}) => {

  const isMultiple = timeslotsCount > 1;

  const price = isMultiple ? (
    <div>
      £{numeral(minPrice).format('0,0')} – £{numeral(maxPrice).format('0,0')}
    </div>
  ) : (
    <span>£{numeral(minPrice).format('0,0')}</span>
  );

  const icon = cloudinary.url(typeIcon, { width: 16, height: 16 });

  return (
    <div className={styles.general}>
      <div className={styles.header}>
        <div className={styles.price}>
          {price}
        </div>
      </div>
      <div className={styles.line} />
      <div className={styles.infoWrap}>
        <PlaceIcon className={cn(styles.icon, styles.invertIcon)} />
        <div>{place}</div>
      </div>
      <div className={styles.infoWrap}>
        <img src={icon} alt="" className={styles.icon} />
        <div>{typeName}</div>
      </div>
      <div className={styles.infoWrap}>
        {
          isMultiple ? (
            <TimeslotsTooltipIcon className={styles.icon} />
          ) : (
            <DateIcon className={cn(styles.icon, styles.invertIcon)} />
          )
        }
        <div>{timeslotsCount} Timeslot{isMultiple ? 's' : ''}</div>
      </div>
      <div className={styles.infoWrap}>
        <ResolutionIcon className={cn(styles.icon, styles.invertIcon)} />
        <div>{resolution}px</div>
      </div>
      {
        looping && (
          <div className={styles.infoWrap}>
            <LoopingIcon className={cn(styles.icon, styles.invertIcon)} />
            <div>Looping {looping}</div>
          </div>
        )
      }
    </div>
  );

};

Info.propTypes = {
  place: PropTypes.node,
  typeIcon: PropTypes.string,
  typeName: PropTypes.node,
  timeslotsCount: PropTypes.number,
  minPrice: PropTypes.number,
  maxPrice: PropTypes.number,
  resolution: PropTypes.string,
  looping: PropTypes.string,
};

export default Info;
