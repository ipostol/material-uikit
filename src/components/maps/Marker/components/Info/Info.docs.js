import React from 'react';
import { Skin } from 'react-markup-components';
import Info from './Info';

export default () => (
  <Skin>
    <Info
      timeslotsCount={3}
      minPrice={123}
      maxPrice={321}
      typeIcon="wj0hwqmnqnzj9qgvbisy"
      typeName="Digital Billboard"
      resolution="15x50"
      looping="30s, 30m, 10m, 10m, 30s, 30m, 10m, 10m"
      place="Gregg Point"
    />
  </Skin>
);
