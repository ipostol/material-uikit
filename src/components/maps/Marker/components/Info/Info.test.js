import React from 'react';
import { shallow } from 'enzyme';
import Info from './Info';

describe('Info component', () => {

  it('should render correctly', () => {

    const props = {
      looping: 'looping',
    };

    const wrapper = shallow(
      <Info {...props} />,
    );

    expect(wrapper).toMatchSnapshot();

  });

  it('should render correctly timeslotsCount=2', () => {

    const props = {
      timeslotsCount: 2,
    };

    const wrapper = shallow(
      <Info {...props} />,
    );

    expect(wrapper).toMatchSnapshot();

  });

});
