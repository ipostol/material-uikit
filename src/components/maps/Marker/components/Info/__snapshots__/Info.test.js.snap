// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info component should render correctly 1`] = `
<div
  className="general"
>
  <div
    className="header"
  >
    <div
      className="price"
    >
      <span>
        £
        0
      </span>
    </div>
  </div>
  <div
    className="line"
  />
  <div
    className="infoWrap"
  >
    <Place
      className="icon invertIcon"
    />
    <div />
  </div>
  <div
    className="infoWrap"
  >
    <img
      alt=""
      className="icon"
    />
    <div />
  </div>
  <div
    className="infoWrap"
  >
    <Timeslots
      className="icon invertIcon"
    />
    <div>
       Timeslot
    </div>
  </div>
  <div
    className="infoWrap"
  >
    <Resolution
      className="icon invertIcon"
    />
    <div>
      px
    </div>
  </div>
  <div
    className="infoWrap"
  >
    <Looping
      className="icon invertIcon"
    />
    <div>
      Looping 
      looping
    </div>
  </div>
</div>
`;

exports[`Info component should render correctly timeslotsCount=2 1`] = `
<div
  className="general"
>
  <div
    className="header"
  >
    <div
      className="price"
    >
      <div>
        £
        0
         – £
        0
      </div>
    </div>
  </div>
  <div
    className="line"
  />
  <div
    className="infoWrap"
  >
    <Place
      className="icon invertIcon"
    />
    <div />
  </div>
  <div
    className="infoWrap"
  >
    <img
      alt=""
      className="icon"
    />
    <div />
  </div>
  <div
    className="infoWrap"
  >
    <TimeslotsTooltip
      className="icon"
    />
    <div>
      2
       Timeslot
      s
    </div>
  </div>
  <div
    className="infoWrap"
  >
    <Resolution
      className="icon invertIcon"
    />
    <div>
      px
    </div>
  </div>
</div>
`;
